fix: Dockerized pre-commit-terraform (#219)
antonbabenko/pre-commit-terraform: README.md

### 1. Install dependencies

<!-- markdownlint-disable no-inline-html -->

* [`pre-commit`](https://pre-commit.com/#install),
  <sub><sup>[`terraform`](https://www.terraform.io/downloads.html),
  <sub><sup>[`git`](https://git-scm.com/downloads),
  <sub><sup>POSIX compatible shell,
  <sub><sup>Internet connection (on first run),
  <sub><sup>x86_64 compatible operation system,
  <sub><sup>Some hardware where this OS will run,
  <sub><sup>Electricity for hardware and internet connection,
  <sub><sup>Some basic physical laws,
  <sub><sup>Hope that it all will works.
  </sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub><br><br>
* [`checkov`](https://github.com/bridgecrewio/checkov) required for `checkov` hook.
* [`terraform-docs`](https://github.com/terraform-docs/terraform-docs) required for `terraform_docs` hooks.
* [`terrascan`](https://github.com/accurics/terrascan) required for `terrascan` hook.
* [`TFLint`](https://github.com/terraform-linters/tflint) required for `terraform_tflint` hook.
* [`TFSec`](https://github.com/liamg/tfsec) required for `terraform_tfsec` hook.

<details><summary><b>Docker</b></summary><br>

> If `--build-arg`'s not specified - will be installed only latests version of `pre-commit` and `terraform`.

Reviewer: ```suggestion
If no `--build-arg` is specified, then the latest versions of `pre-commit` and `terraform` will be installed.
```

* [`checkov`](https://github.com/bridgecrewio/checkov) required for `checkov` hook.
* [`terraform-docs`](https://github.com/terraform-docs/terraform-docs) required for `terraform_docs` hooks.
* [`terrascan`](https://github.com/accurics/terrascan) required for `terrascan` hook.
* [`TFLint`](https://github.com/terraform-linters/tflint) required for `terraform_tflint` hook.
* [`TFSec`](https://github.com/liamg/tfsec) required for `terraform_tfsec` hook.

<details><summary>Docker</summary>

```bash
git clone git@github.com:antonbabenko/pre-commit-terraform.git
cd pre-commit-terraform
docker build -t pre-commit .
```

You can specify needed tool versions by providing `build-args`.  
If `build-args` not specified - will be used default versions from [Dockerfile](Dockerfile)

```bash
docker build -t pre-commit \
    --build-arg PRE_COMMIT_VERSION=2.11.1 \
    --build-arg TERRAFORM_VERSION=0.15.0 \
    --build-arg CHECKOV_VERSION=1.0.838 \
    --build-arg TERRAFORM_DOCS_VERSION=0.12.0 \
    --build-arg TERRASCAN_VERSION=1.10.0 \
    --build-arg TFLINT_VERSION=0.27.0 \
    --build-arg TFSEC_VERSION=0.58.6 \
    .
```

> If you'd like to use the latest versions - you are able to do this too

Reviewer: Let's change it so that users should pass values for optional utilities (`TFSEC_VERSION`, `CHECKOV_VERSION`,`TERRASCAN_VERSION`,`TFLINT_VERSION`) if they want to install such tool. If they don't put version or `latest` then there will be no such tool for them.

The required one is `pre-commit` and `terraform`.
